query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryServiceImpl.java

```diff
@@ -30,41 +28,25 @@ public QueryServiceImpl(QueryServiceImplConfig config) {
         config.getClients().stream()
             .map(ClientConfig::parse)
             .collect(Collectors.toMap(ClientConfig::getType, clientConfig -> clientConfig));
-    String tenantColumnName = config.getTenantColumnName();
-
-    if (tenantColumnName == null || tenantColumnName.isBlank()) {
-      throw new RuntimeException(
-          "Tenant column name is not defined. Need to set service.config.tenantColumnName in the application config.");
-    }
 
     for (Config requestHandlerConfig : config.getQueryRequestHandlersConfig()) {
       initRequestHandler(
-          RequestHandlerConfig.parse(requestHandlerConfig), clientConfigMap, tenantColumnName);
+          RequestHandlerConfig.parse(requestHandlerConfig), clientConfigMap);
     }
     selector = new RequestHandlerSelector(RequestHandlerRegistry.get());
   }
 
-  private void initRequestHandler(
-      RequestHandlerConfig config,
-      Map<String, ClientConfig> clientConfigMap,
-      String tenantColumnName) {
+  private void initRequestHandler(RequestHandlerConfig config,
+      Map<String, ClientConfig> clientConfigMap) {
 
     // Register Pinot RequestHandler
     if ("pinot".equals(config.getType())) {
-      Map<String, Object> requestHandlerInfoConf = new HashMap<>();
-      requestHandlerInfoConf.put(
-          PinotBasedRequestHandler.VIEW_DEFINITION_CONFIG_KEY,
-          ViewDefinition.parse(
-              (Map<String, Object>)
-                  config
-                      .getRequestHandlerInfo()
-                      .get(PinotBasedRequestHandler.VIEW_DEFINITION_CONFIG_KEY),
-              tenantColumnName));
-      RequestHandlerRegistry.get()
-          .register(
-              config.getName(),
-              new RequestHandlerInfo(
-                  config.getName(), PinotBasedRequestHandler.class, requestHandlerInfoConf));
+      boolean registered = RequestHandlerRegistry.get().register(config.getName(),
```

Comment:
This is hard to test?

Reply:
Currently the way PinotClient is created and first connections are made, yes, it's not straight forward. Maybe in a next PR.